File: Gnome sort/Gnome sort-Javascript/gnome_sort.js
const gnomeSort = (arr) => {
    const moveBack = (i) => {
        for( ; i > 0 && arr[i-1] > arr[i]; i--) {
            var t = arr[i];
            arr[i] = arr[i-1];
            arr[i-1] = t;
        }
    }

    for (var i = 1; i < arr.length; i++) {
        if (arr[i-1] > arr[i]) moveBack(i);
    }
    return arr;
}

let arr = [3, 0, 2, 5, -1, 4, 1];
gnomeSort(arr)
console.log(arr);

File: Gnome sort/GnomeSort_Java/GnomeSort.java
package otros.algoritmos;

import java.util.Arrays;

public class GnomeSort {
	
	static void gnomeSort(int arr[], int x) {
		int cont = 0;
		while(cont < x) {
			if (cont == 0) {
				cont++;
			} else if (arr[cont] >= arr[cont - 1]) {
				cont++;
			} else {
				int tmp = 0;
				tmp = arr[cont];
				arr[cont] = arr[cont - 1];
				arr[cont - 1] = tmp;
				cont--;
			}
		}
		return;
	}

	public static void main(String[] args) {
		int arr[] = { 345, 123, 34, 45, -15, -89, 0, -124};
		gnomeSort(arr, arr.length);
		System.out.print(Arrays.toString(arr));

	}

}
